### PR TITLE
App theming. Part 1

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)
- Refactors the codebase, replacing inline styling with global styling. More refactoring to happen in Part 2
- Reduces constants used in database setup.
- Improves  typography.

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Follow instructions in the [readme.md](https://github.com/airqo-platform/AirQo-frontend/tree/app-theming/mobile#airqo-mobile-app)

#### Screenshots (optional)

### CODE REVIEW

File: mobile/lib/auth/change_email_screen.dart [hunk 1]
```diff
@@ -218,15 +218,15 @@ class ChangeEmailScreenState extends State<ChangeEmailScreen> {
     ));
   }
 
-  void clearEmailCallBack() {
+  void _clearEmailCallBack() {
```
Comment (NoahNsimbe) [inline mobile/lib/auth/change_email_screen.dart:221]: making functions private

File: mobile/lib/auth/email_auth_widget.dart [hunk 2]
```diff
@@ -273,54 +83,155 @@ class EmailAuthWidgetState extends State<EmailAuthWidget> {
...
+      controller: _emailInputController,
+      onTap: () {
+        setState(() {
+          _showAuthOptions = false;
+        });
+      },
+      onEditingComplete: () async {
+        FocusScope.of(context).requestFocus(FocusNode());
+        Future.delayed(const Duration(milliseconds: 250), () {
+          setState(() {
```
Comment (NoahNsimbe) [inline mobile/lib/auth/email_auth_widget.dart:96]: code cleanup on the email input field

File: mobile/lib/auth/phone_auth_widget.dart [hunk 3]
```diff
@@ -427,72 +382,214 @@ class PhoneAuthWidgetState extends State<PhoneAuthWidget> {
...
+      const SizedBox(
+        height: 8,
+      ),
+      Padding(
+        padding: const EdgeInsets.only(left: 40, right: 40),
+        child: AutoSizeText(
+            // 'Enter the 6 digits code sent to your\n'
+            //     'number that ends with ...'
+            //     '${phoneNumber.substring(phoneNumber.length - 3)}',
+            'Enter the 6 digits code sent to your '
```
Comment (NoahNsimbe) [inline mobile/lib/auth/phone_auth_widget.dart:409]: temporarily disabled hiding of phone numbers

File: mobile/lib/themes/light_theme.dart [hunk 4]
```diff
@@ -260,17 +260,78 @@ const googleMapsLightTheme = [
 ThemeData lightTheme() {
   final base = ThemeData.light();
 
+  /// height := line height * fontSize
+  /// letterSpacing := 16 * letter-spacing(em)
+  /// Theme.of(context).textTheme.headline1
+  /// CustomTextStyle.headline10(context)
+
```
Comment (NoahNsimbe) [inline mobile/lib/themes/light_theme.dart:267]: short notes on how to use the styling

File: mobile/lib/themes/light_theme.dart [hunk 5]
```diff
@@ -260,17 +260,78 @@ const googleMapsLightTheme = [
...
     // accentColor: Config.appColor,
     // backgroundColor: Colors.white,
     visualDensity: VisualDensity.adaptivePlatformDensity,
-    textTheme: ThemeData.light().textTheme.apply(
-          fontFamily: 'Inter',
-          bodyColor: Config.appColor,
-          displayColor: Config.appColor,
-        ),
-    // canvasColor: Colors.transparent,
+    textTheme: TextTheme(
```
Comment (NoahNsimbe) [inline mobile/lib/themes/light_theme.dart:273]: overriding default styles

File: mobile/lib/themes/light_theme.dart [hunk 6]
```diff
@@ -282,3 +343,99 @@ ThemeData lightTheme() {
     appBarTheme: const AppBarTheme(elevation: 0.0),
   );
 }
+
+class CustomTextStyle {
```
Comment (NoahNsimbe) [inline mobile/lib/themes/light_theme.dart:347]: Created a custom style class to handle additional styles specified in figma

File: mobile/lib/widgets/custom_widgets.dart [hunk 7]
```diff
@@ -226,30 +300,32 @@ Widget searchLocationTile(Measurement measurement) {
         border: Border.all(color: Colors.transparent)),
     child: ListTile(
       contentPadding: const EdgeInsets.only(left: 0.0),
-      title: Text(
+      title: AutoSizeText(
         measurement.site.name,
         maxLines: 1,
         overflow: TextOverflow.ellipsis,
-        style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 16),
+        style: CustomTextStyle.headline8(context),
```
Comment (NoahNsimbe) [inline mobile/lib/widgets/custom_widgets.dart:307]: Using a global custom text style

File: mobile/lib/widgets/insights_tab.dart [hunk 8]
```diff
@@ -95,8 +96,8 @@ class _InsightsTabState extends State<InsightsTab> {
                       visible: _hasMeasurements,
                       child: Text(
                         'AIR QUALITY'.toUpperCase(),
-                        style: TextStyle(
-                            fontSize: 12, color: Colors.black.withOpacity(0.3)),
+                        style: Theme.of(context).textTheme.caption?.copyWith(
```
Comment (NoahNsimbe) [inline mobile/lib/widgets/insights_tab.dart:99]: Using global styles

**Review by 123MwanjeMike — APPROVED** [review]

**Review by Baalmart — APPROVED** [review]
LGTM

**Review by Baalmart — APPROVED** [review]
All looks good.

**Pull request merged**